cserver.py

```diff
@@ -317,6 +318,50 @@ def query():
             items = q.items
             for i in items:
                 res.append({'key': i.key, 'value': i.value})
+
+    elif string == 'get_user':
+      search_input = request.args.get('search_input')
+
+      if not search_input:
+        res.append('error')
+
+      else:
+        all_enrollments = Kv.query.filter(and_(
+          Kv.testnet == testnet,
+          Kv.value.like('%\nRein User Enrollment%')
+        ))
+
+        # Check for SIN matches
+        q = all_enrollments.filter(
+          Kv.value.like('%\nSecure Identity Number: {}%'.format(search_input))
+        ).paginate(1, 20, False)
+
+        # If unsuccessful, check for master address matches
+        if not q.items:
+          q = all_enrollments.filter(
+            Kv.value.like('%\nMaster signing address: {}%'.format(search_input))
+          ).paginate(1, 20, False)
+
+        # If unsuccessful, check for delegate address matches
+        if not q.items:
+          q = all_enrollments.filter(
+            Kv.value.like('%\nDelegate signing address: {}%'.format(search_input))
+          ).paginate(1, 20, False)
+
+        # If unsuccessful, check for username, case-insensitive, accepts partial matches
+        if not q.items:
+          q = all_enrollments.filter(
+            Kv.value.ilike('%\nUser: %{}%'.format(search_input))
```

Comment:
The middle % here might match any number of newlines so I think we end up searching all text in the enrollment after "User: ". For example, "Mediator" and "signature" both pull up all three users I have in my test instance. If you can think of a simple fix, go for it. Otherwise, leave it and we'll optimize later.